### ISSUE TITLE
Date error in b4 on Sonoma v14.0

### ISSUE THREAD

**Issue by ofirgalcon** [Macjutsu/super#156]
`**** S.U.P.E.R.M.A.N. 4.0.0-beta4 - SUPER STARTUP WORKFLOW ****
Failed conversion of ``12 Oct 14:41:00'' using format ``%b %d %H:%M:%S''
date: illegal time format
usage: date [-jnRu] [-I[date|hours|minutes|seconds]] [-f input_fmt]
            [-r filename|seconds] [-v[+|-]val[y|m|w|d|H|M|S]]
            [[[[mm]dd]HH]MM[[cc]yy][.SS] | new_date] [+output_fmt]
Status: Mac computer with Intel running: macOS Sonoma 14.0-23A344`

**Comment by ofirgalcon** [comment]
The output of `last -1 reboot
reboot time                                Thu 12 Oct 14:41`

UK format seems to break the code

**Comment by ofirgalcon** [comment]
Error is at line 3416

`mac_last_startup=$(date -j -f "%b %d %H:%M:%S" "${last_reboot_response}:00" +"%Y-%m-%d:%H:%M:%S") # Expected output: 2023-08-25:00:16:00
`

**Comment by Macjutsu** [comment]
ugh... what version of macOS are you on? I thought I had this sorted.

Different versions of macOS return different values.

**Comment by Macjutsu** [comment]
Ok... I'm gonna try a different approach... can you try this line please:

`last -1 reboot | sed -e 's/  */ /g' | awk '{print $4" "$5" "$6;}'`

it should return a format like this: `Sep 28 21:21`

**Comment by ofirgalcon** [comment]
I get `12 Oct 14:41` on UK format

**Comment by Macjutsu** [comment]
ugh.... that makes this so much more complex

**Comment by ofirgalcon** [comment]
I know, I tried fixing it myself quickly and failed.

**Comment by Macjutsu** [comment]
hrm... I think I'm gonna detect if the first character is a number or letter and then assume there are "only" two formats.

**Comment by Macjutsu** [comment]
@ofirgalcon Can you try these two commands please?

`last_reboot_response=$(last -1 reboot | sed -e 's/  */ /g' | awk '{print $4" "$5" "$6;}')`

`date -j -f "%d %b %H:%M:%S" "${last_reboot_response}:00" +"%Y-%m-%d:%H:%M:%S"`

Output should look like this: `2023-08-25:00:16:00`

**Comment by ofirgalcon** [comment]
```
Failed conversion of ``o o t:00'' using format ``%d %b %H:%M:%S''
date: illegal time format
usage: date [-jnRu] [-I[date|hours|minutes|seconds]] [-f input_fmt]
            [-r filename|seconds] [-v[+|-]val[y|m|w|d|H|M|S]]
            [[[[mm]dd]HH]MM[[cc]yy][.SS] | new_date] [+output_fmt]
```

when I add `echo "${last_reboot_response}"`

I get

`o o t`

**Comment by Macjutsu** [comment]
yuk... what does just `last -1 reboot` put out?

**Comment by ofirgalcon** [comment]
`reboot time                                Thu 12 Oct 14:41`

**Comment by Macjutsu** [comment]
Can you test this please?
`last -1 reboot | sed -e 's/  */ /g' | awk '{print $4" "$5" "$6;}'`
To be clear... running as root/sudo.

**Comment by ofirgalcon** [comment]
```
last -1 reboot | sed -e 's/ */ /g' | awk '{print $4" "$5" "$6;}'

o o t 
```

**Comment by Macjutsu** [comment]
OK... so this is a github problem... I can use that same snippet of code directly from BBedit and it works... but the code I just copied and pasted from github doesn't work. They must be sneaking in some invisibles in there.

**Comment by Macjutsu** [comment]
https://github.com/Macjutsu/super/releases/tag/v4.0.0-beta6

**Comment by ofirgalcon** [comment]
I get this error when I run b6

`Error: Unregonized last reboot date format: 12 Oct 14:41`

**Comment by Macjutsu** [comment]
Of course... b/c why can't any of these methods have the same freaking time format?

Anyhow... this is clearly a localization issue... so If you could please replace the following if statement starting on line 3541 and lemme know how that goes... thanks again!

```
if [[ "${last_reboot_response}" =~ ^[a-zA-Z]{3}\ [0-9]{2}\ [0-9]{2}:[0-9]{2}$ ]]; then
	mac_last_startup=$(date -j -f "%b %d %H:%M:%S" "${last_reboot_response}:00" +"%Y-%m-%d:%H:%M:%S") # Expected output: 2023-08-25:00:16:00
elif [[ "${last_reboot_response}" =~ ^[0-9]{2}\ [a-zA-Z]{3}\ [0-9]{2}:[0-9]{2}$ ]]; then
	mac_last_startup=$(date -j -f "%d %b %H:%M:%S" "${last_reboot_response}:00" +"%Y-%m-%d:%H:%M:%S") # Expected output: 2023-08-25:00:16:00
else # Unknown format.
	log_super "Error: Unregonized last reboot date format: ${last_reboot_response}"; check_error="TRUE"
fi
```

**Comment by ofirgalcon** [comment]
I think it's working now.

**Comment by Macjutsu** [comment]
@bibliocommons-github

If you find your way back here please try this latest version... replace the if statement starting on line 3541 with this...

```
if [[ "${last_reboot_response}" =~ ^[a-zA-Z]{3}\ [0-9]+\ [0-9]{2}:[0-9]{2}$ ]]; then
	mac_last_startup=$(date -j -f "%b %d %H:%M:%S" "${last_reboot_response}:00" +"%Y-%m-%d:%H:%M:%S") # Expected output: 2023-08-25:00:16:00
elif [[ "${last_reboot_response}" =~ ^[0-9]+\ [a-zA-Z]{3}\ [0-9]{2}:[0-9]{2}$ ]]; then
	mac_last_startup=$(date -j -f "%d %b %H:%M:%S" "${last_reboot_response}:00" +"%Y-%m-%d:%H:%M:%S") # Expected output: 2023-08-25:00:16:00
else # Unknown format.
	log_super "Error: Unrecognized last reboot date format: ${last_reboot_response}"; check_error="TRUE"
fi
```
